### PR TITLE
Use structured output from sync command

### PR DESCRIPTION
**Pull request by pietern**
This removes the need for complex state tracking and monitors for "start" and "complete" events in the sync command stdout. As we no longer forward log messages to the VS code terminal pane this change includes rudimentary formatting of these messages into something human readable.

We should consider if it is worth keeping this pane or if the status should be consolidated into the "sync" pane.

### CODE REVIEW

File: packages/databricks-vscode/src/cli/BricksSyncParser.ts [hunk 1]
```diff
@@ -103,38 +106,23 @@ export class BricksSyncParser {
...
+    // This function processes the JSON output from bricks sync and parses it
+    // to figure out if a synchronization step is in progress or has completed.
+    public processStdout(data: string) {
+        const logLines = data.split("\n");
+        for (let i = 0; i < logLines.length; i++) {
+            const line = logLines[i];
+            try {
+                this.processLine(line);
+            } catch (error: any) {
+                // Log error?
```
Comment (pietern) [inline packages/databricks-vscode/src/cli/BricksSyncParser.ts:122]: @kartikgupta-db Which logger should I use here?

**Comment by pietern** [comment]
Note: tests will fail because a bump is required.

File: packages/databricks-vscode/src/cli/BricksSyncParser.ts (hunk 1 above)
Comment (kartikgupta-db) [reply packages/databricks-vscode/src/cli/BricksSyncParser.ts:122]: @pietern  Please use the extension logger here. 
`NamedLogger.getOrCreate(Loggers.Extension)`

**Comment by fjakobs** [comment]
A little video is always appreciated

**Review by fjakobs — APPROVED** [review]

File: packages/databricks-vscode/src/cli/BricksSyncParser.ts (hunk 1 above)
Comment (pietern) [reply packages/databricks-vscode/src/cli/BricksSyncParser.ts:122]: Thanks, done.

**Pull request merged**